google/cloud/pubsub/internal/pull_message.h

```diff
@@ -0,0 +1,46 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_PUBSUB_INTERNAL_PULL_MESSAGE_H
+#define GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_PUBSUB_INTERNAL_PULL_MESSAGE_H
+
+#include "google/cloud/pubsub/message.h"
+
+namespace google {
+namespace cloud {
+namespace pubsub_internal {
+GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
+
+/**
+ * An interface for acting on a message received via the Unary Pull in
+ * `SubscriberConnectionImpl`.
+ *
+ * The `SubscriberConnectionImpl` receives a single message. This interface
+ * offers a way to trace the life of the message.
```

Comment:
Why do we need that?  Typically the interesting thing is the lifetime of the `AckHandler`.  If the application makes a copy of `pubsub::Message` but already `ack/nacked` the message then we are done.  The rest of the lifetime is their problem.

Reply:
Initially the design included stripping the OTel related message attributes before giving it to the user and linking the settle span to the create span but that has changed, so I don't think we need this anymore.